Update design docs instructions (#4183)
ros2/ros2_documentation: source/The-ROS2-Project/Contributing/Developer-Guide.rst

Writing the design document
~~~~~~~~~~~~~~~~~~~~~~~~~~~

Design docs are a way to document repository or package specific design decisions.
This is different than `ROS Enhancement Proposals (or REPs) <https://ros.org/reps/rep-0000.html>`__, which are for larger decisions that often effect the ROS community.
If you are unsure if you should do a design document or a REP, you can ask the ROS 2 team.

> A good place to ask is in the weekly ROS 2 meeting, or in a Github issue in a relevant ROS 2 repository.

Reviewer: ```suggestion
A good place to ask is to open a GitHub issue in a relevant ROS 2 repository.
```

> Design docs should be placed in a ``doc`` directory in the relevant repository or package.
> And submitted as a pull request that references any related Github issues.
> Make sure to mention any additional context in the pull request such as if the design doc is intended for a specific ROS version.
> Feedback for the design doc wil be made directly on the pull request.

Reviewer: ```suggestion
Design docs should be placed in a ``doc`` directory in the relevant repository or package.
Make sure to mention any additional context in the pull request such as if the design doc is intended for a specific ROS version.
Feedback for the design doc will be made directly on the pull request.
```

> .. note::
>     Design docs should never include confidential information and they are not required for bug fixes.

Reviewer: ```suggestion
.. note::

    Design docs should never include confidential information and they are not required for bug fixes.
```
